envi/archs/ppc/__init__.py

```diff
@@ -192,44 +265,59 @@ def __init__(self, mode=32, archname='ppc-vle'):
     def isVle(self, va):
         return True
 
-    def archParseOpcode(self, bytez, offset=0, va=0):
+    def archParseOpcode(self, bytez, offset=0, va=0, extra=None):
         return self._arch_dis.disasm(bytez, offset, va)
 
     def archGetBreakInstr(self):
-        return '\x7c\x00\x00\xc2'   # e_dnh instruction
+        # BOOKE:        dnh     4C00018C
+        # VLE 32bit:    e_dnh   7C0000C2
+        # VLE 16bit:    se_dnh  000F
+        return '\x7C\x00\x00\xC2'
 
     def archGetNopInstr(self):
-        return '\x18\x00\xd0\x00'   # e_ori 0,0,0 (also could be '\x44\x00' se_or 0,0)
+        # BOOKE:        nop (ori 0,0,0)     60000000
+        # VLE 32bit:    e_nop (e_ori 0,0,0) 1800D000
+        # VLE 16bit:    se_nop (se_or 0,0)  4400
+        return '\x44\x00'
```

Comment:
what are your thoughts on how we have this set up?

i'm not suggesting we change immediately, but do you think VLE deserves it's own architecture?  something like ARM/THUMB?  they share the same emulator, but the architecture alternates appropriately.

again, don't pivot on this right now, but just a thought.  something that would allow each architecture to provide a real NOP and Break instruction.  something that would reduce the amount of data we need to setup/carry in the ENVI ArchModule, but would live in the VivWorkspace and the Emulator.

Reply:
maybe later, or maybe never.  after discussing in DM, this seems like a lot of work that would require changes to the Vivisect core.